src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -242,14 +283,78 @@ public override void ExecuteCmdlet()
                 openPorts: OpenPorts);
             var networkInterface = resourceGroup.CreateNetworkInterfaceConfig(
                 Name, subnet, publicIpAddress, networkSecurityGroup);
-            var virtualMachine = resourceGroup.CreateVirtualMachineConfig(
-                name: Name,
-                networkInterface: networkInterface,
-                isWindows: isWindows,
-                adminUsername: Credential.UserName,
-                adminPassword: new NetworkCredential(string.Empty, Credential.Password).Password,
-                image: image,
-                size: Size);
+            ResourceConfig<VirtualMachine> virtualMachine = null;
+
+            if (image != null)
+            {
+                virtualMachine = resourceGroup.CreateVirtualMachineConfig(
+                    name: Name,
+                    networkInterface: networkInterface,
+                    isWindows: isWindows,
+                    adminUsername: Credential.UserName,
+                    adminPassword: new NetworkCredential(string.Empty, Credential.Password).Password,
+                    image: image,
+                    size: Size);
+            }
+            else
+            {
+                var storageClient =
+                        AzureSession.Instance.ClientFactory.CreateArmClient<StorageManagementClient>(DefaultProfile.DefaultContext,
+                            AzureEnvironment.Endpoint.ResourceManager);
+                var st1 = storageClient.StorageAccounts.Create(ResourceGroupName, Name, new StorageAccountCreateParameters
+                {
```

Comment:
It would be good if call the `Create` function as a part of dependency tree. I can help with this. In the future, we may use generate Azure templates so let's discuss it.

Reply:
I tried, but not trivial. You can check storage account operations. They don't have ```Get``` and they have ```GetProperties```, and they don't have ```CreateOrUpdate``` and they use ```Create```, etc.

Reply:
There is one problem then. A user may specify a new resource group which doesn't exist yet. `SimpleParameterSet` handles this but it will fail in `DiskFileParameterSet`.

Reply:
I think I use the same RG strategy?
```
this ResourceConfig<ResourceGroup> resourceGroup,
```

Reply:
Oh, you mean in the storage creation...

Can we handle it as a bug item? It doesn't seem to be trivial, and it's not an issue for the Simple parameter set, and not a P0 blocker for DiskFile set as well...

Reply:
I'm fine with this as long as we track the issue.

Reply:
Created for tracking: #5279